### PR TITLE
added tests for OfferFilterDrawer, OfferFilterList, fix type errors

### PR DESCRIPTION
**Pull request by D1nviz**
image
image

OfferFilterDrawer can't test for 100% because i can't find the way for adding coverage of this lines

image

### CODE REVIEW

File: tests/unit/containers/find-offer/offer-filter-block/offer-filter-list/OfferFilterList.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+import OfferFilterList from '~/containers/find-offer/offer-filter-block/offer-filter-list/OfferFilterList'
+
+import { useAppSelector } from '~/hooks/use-redux'
+import { ProficiencyLevelEnum, UserRoleEnum } from '~/types'
+import { selectOption } from '~tests/test-utils'
+
+vi.mock('~/hooks/use-redux', () => ({
+  useAppSelector: vi.fn()
+}))
```
Comment (mhevyk) [inline tests/unit/containers/find-offer/offer-filter-block/offer-filter-list/OfferFilterList.spec.jsx:11]: Do we really need to mock redux here and why? Why can't we use `renderWithProviders` test utility with `preloadedState`?

File: tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    renderWithProviders(
+      <CoursesFiltersDrawer
+        additionalParams={{}}
+        filterActions={filterActions}
+        filters={filters}
+        isOpen
+        onClose={mockOnClose}
+      />
+    )
+  })
```
Comment (mhevyk) [inline tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx:55]: I think it should work without `waitFor`

File: tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      fireEvent.click(applyButton)
+      expect(mockUpdateFiltersInQuery).toHaveBeenCalled()
+      expect(mockOnClose).toHaveBeenCalled()
+    })
+  })
+
+  describe('with category filter', () => {
+    beforeEach(async () => {
+      await setup(filterWithCategory)
+    })
```
Comment (mhevyk) [inline tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx:122]: If my previous comment succeeds, remove `async/await` here

File: tests/unit/containers/find-offer/offer-filter-block/offer-filter-list/OfferFilterList.spec.jsx (hunk 1 above)
Comment (D1nviz) [reply tests/unit/containers/find-offer/offer-filter-block/offer-filter-list/OfferFilterList.spec.jsx]: I refactored this code according to your suggestion. Thank you

File: tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx (hunk 2 above)
Comment (D1nviz) [reply tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx:55]: I think this line is required, withour `waitFor` we receiving this warnings in console:

image

File: tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx (hunk 3 above)
Comment (D1nviz) [reply tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx]: Yes, you're right, it reduntend

File: tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx (hunk 2 above)
Comment (mhevyk) [reply tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx:55]: I noticed that a lot of tests don't use waitFor. Maybe we should create a non-urgent issue for it?

Comment (D1nviz) [reply tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx:55]: i think we need to discuss about it on our meeting

**Review by mhevyk — APPROVED** [review]

**Review by GeorgeShvab — APPROVED** [review]

File: src/containers/find-offer/offer-filter-block/offer-filter-list/OfferFilterList.tsx [hunk 4]
```diff
@@ -94,7 +94,7 @@ const OfferFilterList: FC<OfferFilterListProps> = ({
       <CheckboxList
         {...checkboxListProps}
         items={levelOptions}
-        onChange={updateFilterByKey('proficiencyLevel')}
+        onChange={() => updateFilterByKey('proficiencyLevel')}
```
[the remark this answers is not in the document]
Comment (D1nviz) [reply src/containers/find-offer/offer-filter-block/offer-filter-list/OfferFilterList.tsx]: thank you, fixed

File: tests/unit/containers/find-offer/offer-filter-block/offer-filter-list/OfferFilterList.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+describe('OfferFilterList for Tutor', () => {
+  beforeEach(() => {
+    setupWithRole(preloadedTutorState)
+  })
+
+  it('renders filter titles correctly', () => {
+    expect(
+      screen.getByText('findOffers.filterTitles.level')
+    ).toBeInTheDocument()
```
[the remark this answers is not in the document]
Comment (D1nviz) [reply tests/unit/containers/find-offer/offer-filter-block/offer-filter-list/OfferFilterList.spec.jsx]: You're right. Thank you, fixed!

**Review by dmtrth25 — APPROVED** [review]

**Pull request merged**